supervisor/os/data_disk.py

```diff
@@ -216,6 +244,22 @@ async def migrate_disk(self, new_disk: str) -> None:
                     _LOGGER.error,
                 ) from err
 
+            # If migrating from one external data disk to another, rename the old one to prevent conflicts
```

Comment:
Any specific reason to do this here? I'd suggest to do it before `mark_data_move()` and even before `_format_device_with_single_partition()`... That way we got that out of the way, and if it fails, we don't enter the race...

Reply:
oh right because only the filesystem label matters during normal startup. so even though your system will be a bit wonky (external data disk but no partition `hassos-data-external`) it will work fine. Whereas if we do the rename after and it fails then a reboot causes havoc.

Ok will re-arrange, thanks!